jumpscale/packages/marketplace/chats/gitea.py

```diff
@@ -1,38 +1,144 @@
+        }
+        self.solution_metadata.update(metadata)
+        # reserve subdomain
+        subdomain_wid = deployer.create_subdomain(
+            pool_id=self.pool_id,
+            gateway_id=self.gateway.node_id,
+            subdomain=self.domain,
+            addresses=self.addresses,
+            solution_uuid=self.solution_id,
+            **self.solution_metadata,
+        )
+        subdomain_wid = deployer.wait_workload(subdomain_wid, self)
+
+        if not subdomain_wid:
+            raise StopChatFlow(
+                f"Failed to create subdomain {self.domain} on gateway {self.gateway.node_id} {subdomain_wid}"
+            )
+        self.resv_id = deployer.deploy_container(
+            pool_id=self.pool_id,
+            node_id=self.selected_node.node_id,
+            network_name=self.network_view.name,
+            ip_address=self.ip_address,
+            flist=self.FLIST_URL,
+            cpu=self.query["cru"],
+            memory=self.query["mru"] * 1024,
+            env=var_dict,
+            interactive=False,
+            entrypoint="/start_gitea.sh",
+            public_ipv6=True,
+            disk_size=self.query["sru"] * 1024,
+            secret_env={"POSTGRES_PASSWORD": self.database_password},
+            **self.solution_metadata,
+            solution_uuid=self.solution_id,
+        )
+        success = deployer.wait_workload(self.resv_id, self)
+        if not success:
+            solutions.cancel_solution([self.resv_id])
+            raise StopChatFlow(f"Failed to deploy workload {self.resv_id}")
+
+        self.proxy_id = deployer.create_proxy(
+            pool_id=self.pool_id,
+            gateway_id=self.gateway.node_id,
+            domain_name=self.domain,
+            trc_secret=self.secret,
+            **self.solution_metadata,
+            solution_uuid=self.solution_id,
+        )
+        success = deployer.wait_workload(self.proxy_id, self)
+        if not success:
+            solutions.cancel_solution([self.proxy_id])
+            raise StopChatFlow(f"Failed to reserve reverse proxy workload {self.proxy_id}")
+
+        self.tcprouter_id = deployer.expose_address(
+            pool_id=self.pool_id,
+            gateway_id=self.gateway.node_id,
+            network_name=self.network_view.name,
+            local_ip=self.ip_address,
+            port=3000,
+            tls_port=3000,
+            trc_secret=self.secret,
+            bot=self,
+            **self.solution_metadata,
+            solution_uuid=self.solution_id,
+        )
+        success = deployer.wait_workload(self.tcprouter_id)
+        if not success:
+            solutions.cancel_solution([self.tcprouter_id])
+            raise StopChatFlow(f"Failed to reserve tcprouter container workload {self.tcprouter_id}")
+        self.container_url_https = f"https://{self.domain}"
+
+    @chatflow_step(title="Success", disable_previous=True, final_step=True)
+    def success(self):
```

Comment:
there is a general success step in `MarketPlaceAppsChatflow` class, you can remove this implementation.

Reply:
or at least use it to update the message...

Reply:
I modified the message to follow the template in the `MarketPlaceAppsChatflow` class.